Frontends/glslang/GlslangToTopVisitor.cpp

```diff
@@ -1391,7 +1391,12 @@ bool TGlslangToTopTraverser::visitAggregate(glslang::TVisit visit, glslang::TInt
         result = createUnaryIntrinsic(node->getOp(), precision, operands.front());
         break;
     default:
-        result = createIntrinsic(node->getOp(), precision, uniform, operands, glslangOperands.front()->getAsTyped()->getBasicType() == glslang::EbtUint);
+        // Check first for intrinsics that can be done natively
+        if (node->getOp() == glslang::EOpMix && 
+              gla::GetBasicTypeID(operands.front()) == llvm::Type::IntegerTyID)
+            result = llvmBuilder.CreateSelect(operands[2], operands[1], operands[0]);
+        else
+            result = createIntrinsic(node->getOp(), precision, uniform, operands, glslangOperands.front()->getAsTyped()->getBasicType() == glslang::EbtUint);
```

Comment:
It seems strange to have a switch statement to do op-specific work, but then not use that and special case this here instead.  Can you set an LLVM op above in a `case glslang::EOpMix:` and fix operand order there?  Will certainly want to do that as soon as there are more of these.  Though I see it might appear too cumbersome.  (I'm assuming an LLVM create op that takes a vector of operands to make this go smoothly.)